InnerEye/ML/Histopathology/models/deepmil.py

```diff
@@ -179,12 +179,13 @@ def get_metrics(self) -> nn.ModuleDict:
                                   MetricsKey.ACC_WEIGHTED: Accuracy(num_classes=self.n_classes, average='weighted'),
                                   MetricsKey.CONF_MATRIX: ConfusionMatrix(num_classes=self.n_classes)})
         else:
-            return nn.ModuleDict({MetricsKey.ACC: Accuracy(),
+            threshold = 0.5
+            return nn.ModuleDict({MetricsKey.ACC: Accuracy(threshold=threshold),
                                   MetricsKey.AUROC: AUROC(num_classes=self.n_classes),
-                                  MetricsKey.PRECISION: Precision(),
-                                  MetricsKey.RECALL: Recall(),
-                                  MetricsKey.F1: F1(),
-                                  MetricsKey.CONF_MATRIX: ConfusionMatrix(num_classes=self.n_classes+1)})
+                                  MetricsKey.PRECISION: Precision(threshold=threshold),
+                                  MetricsKey.RECALL: Recall(threshold=threshold),
+                                  MetricsKey.F1: F1(threshold=threshold),
+                                  MetricsKey.CONF_MATRIX: ConfusionMatrix(num_classes=2, threshold=threshold)})
```

Comment:
before it was `num_classes=self.n_classes + 1`, was this change intentional?

Reply:
This block is executed only when `n_classes == 1` (i.e. binary) so I preferred to make this explicit.

Reply:
I see now. I think the fact self.n_classes=1 and num_classes=2 are equivalent cases is a bit confusing. We might want to revise this for clarity at a certain point